jsapp/js/components/processing/analysis/editors/analysisQuestionEditor.component.tsx

```diff
@@ -134,13 +134,22 @@ export default function AnalysisQuestionEditor(
           updatedQuestions
         );
 
-        // Step 4: update reducer's state with new list after the call finishes
-        analysisQuestions?.dispatch({
-          type: 'updateQuestionCompleted',
-          payload: {
-            questions: getQuestionsFromSchema(response?.advanced_features),
-          },
-        });
+        // We get all questions in the response, but we only need the one we've
+        // just updated
+        const newQuestions = getQuestionsFromSchema(response?.advanced_features);
+        const currentNewQuestion = newQuestions.find((item) => item.uuid === props.uuid);
+
+        if (currentNewQuestion) {
+          // Step 4: update reducer's state with new list after the call finishes
+          analysisQuestions?.dispatch({
+            type: 'updateQuestionCompleted',
+            payload: {question: currentNewQuestion},
+          });
+        } else {
+          // This should never happen :) I.e. the list of questions from
+          // `response` will include the question, it's just the `.find`
+          // that has a possibility to return `undefined` :shrug:
```

Comment:
I know this should in theory be an unreachable code path, but would it make sense to just `throw` an error here so it bubbles up to the `catch` block?